manifest/provider/validate.go

```diff
@@ -117,6 +117,39 @@ func (s *RawProviderServer) ValidateResourceTypeConfig(ctx context.Context, req
 		}
 	}
 
+	// validate wait block
+	if wait, ok := configVal["wait"]; ok && !wait.IsNull() {
+		var waitBlock []tftypes.Value
+		wait.As(&waitBlock)
+		if len(waitBlock) > 0 {
+			var w map[string]tftypes.Value
+			waitBlock[0].As(&w)
+			n := 0
+			for _, ww := range w {
+				if !ww.IsNull() {
+					n += 1
+				}
+			}
+			if n > 1 {
+
+				resp.Diagnostics = append(resp.Diagnostics, &tfprotov5.Diagnostic{
```

Comment:
This could just go in place of `n += 1` above, followed by a break statement to end the loop. N variable wouldn't be needed in that case.